lms/djangoapps/branding_by_referer/configuration_by_referer.py

```diff
@@ -0,0 +1,161 @@
+            if isinstance(value, dict):
+                try:
+                    merged = current_conf_values.get(key, {}).copy()
+                except AttributeError:
+                    merged = {}
+                merged.update(value)
+                current_conf_values[key] = merged
+                continue
+            current_conf_values[key] = value
+
+        return
+
+    def _update_conf(self, request, referer_domain):
+        """
+        Update the current site configuration based on the referer
+        """
+        # Adding a key to the current configuration to identify it as a overriden set
+        current_conf = configuration_helpers.get_current_site_configuration()
+        current_conf.values[self.OVERRIDE_MKTG_REFERER_KEY] = referer_domain
+
+        self._override_marketing_urls(referer_domain)
+        self._insert_referer_configurations(referer_domain)
+        return
+
+    def get_stored_referer_preference(self, request):
+        """
+        Read the referer domain value from the current logged-in user preferences
+        """
+
+        if not request.user.is_authenticated():
+            return None
+
+        stored_referer_data = UserPreference.get_value(request.user, self.MARKETING_SITE_REFERER)
+        if not stored_referer_data:
+            return None
+
+        try:
+            stored_referer_data_json = json.loads(stored_referer_data)
+            referer_domain_preference = stored_referer_data_json['referer_domain']
+        except ValueError:
+            # TODO: log something
+            return None
+
+        return referer_domain_preference
+
+    def process_response(self, request, response):
+        """
+        Process response middleware method.
+        """
+        possible_override_referer = configuration_helpers.get_value(self.OVERRIDE_MKTG_REFERER_KEY)
+        # if the override referer is present, clean the configuration
+        if possible_override_referer:
+            current_conf = configuration_helpers.get_current_site_configuration()
+            setattr(current_conf, 'values', request.original_site_conf)
+            request.original_site_conf = None
+        return response
+
+    def is_feature_enabled(self):
+        """
+        Check if the ENABLE_BRANDING_BY_REFERER feature flag is set and return its value.
+        """
+        return configuration_helpers.get_value('FEATURES', {}).get('ENABLE_CONFIGURATION_BY_REFERER', False)
+
+    def validate_reset_configuration(self, request):
+        """
+        As a security mesure, make sure the configuration was reset and has no overrides
+        """
+        possible_override_referer = configuration_helpers.get_value(self.OVERRIDE_MKTG_REFERER_KEY)
+        # if this setting is present, we need to reload the current settings as they were not correctly cleaned
+        # in the last request
+        if possible_override_referer:
+            current_conf = configuration_helpers.get_current_site_configuration()
```

Comment:
you could do all on one line on a try catch and be done with the if logic

Reply:
Done